File: screen/model/fileHandler.js
const fs = require('fs');
const path = require('path');
const util = require('util');

const writeFile = util.promisify(fs.writeFile);
const mkdir = util.promisify(fs.mkdir);

module.exports = class FileHandler {

    constructor(resourcesPath) {
        this.resourcesPath = resourcesPath;
    }

    async writeFile(message) {
        const { path: filePath, data } = message;
        const fullPath = path.join(this.resourcesPath, filePath);
        const dir = path.dirname(fullPath);
        mkdir(dir, { recursive: true });
        const buffer = Buffer.from(data, 'base64');
        await writeFile(fullPath, buffer);
        console.log(`Wrote file ${filePath}`);
    }

};

File: audio/src/file_handler.py
from pathlib import Path

class FileHandler:

    def __init__(self, resource_path: Path):
        self._resource_path = resource_path

    def write_file(self, path: Path, data: bytes):
        full_path = self._resource_path / path
        full_path.parent.mkdir(parents=True, exist_ok=True)
        with open(full_path, "wb") as f:
            f.write(data)
        print("Wrote file " + str(path))